src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mikkelinyt/MikkeliNytEntityUpdater.java

```diff
@@ -0,0 +1,286 @@
+  
+  @Resource
+  private TimerService timerService;
+
+  private boolean stopped;
+  private List<OrganizationId> queue;
+
+  @PostConstruct
+  public void init() {
+    queue = Collections.synchronizedList(new ArrayList<>());
+  }
+
+  @Override
+  public String getName() {
+    return "mikkeli-nyt-events";
+  }
+
+  @Override
+  public void startTimer() {
+    startTimer(TIMER_INTERVAL);
+  }
+
+  private void startTimer(int duration) {
+    stopped = false;
+    TimerConfig timerConfig = new TimerConfig();
+    timerConfig.setPersistent(false);
+    timerService.createSingleActionTimer(duration, timerConfig);
+  }
+
+  @Override
+  public void stopTimer() {
+    stopped = true;
+  }
+  
+  @Asynchronous
+  public void onOrganizationIdUpdateRequest(@Observes OrganizationIdUpdateRequest event) {
+    if (!stopped) {
+      OrganizationId organizationId = event.getId();
+      if (getApiKey(organizationId) == null)  {
+        return;
+      }
+      
+      if (event.isPriority()) {
+        queue.remove(organizationId);
+        queue.add(0, organizationId);
+      } else {
+        if (!queue.contains(organizationId)) {
+          queue.add(organizationId);
+        }
+      }
+    }
+  }
+
+  @Timeout
+  public void timeout(Timer timer) {
+    if (!stopped) {
+      if (!queue.isEmpty()) {
+        updateEvents(queue.remove(0));
+      }
+
+      startTimer(SystemUtils.inTestMode() ? 1000 : TIMER_INTERVAL);
+    }
+  }
+
+  private void updateEvents(OrganizationId organizationId) {
+    Response<EventsResponse> response = listEvents(organizationId);
+    if (response.isOk()) {
+      for (fi.otavanopisto.mikkelinyt.model.Event event : response.getResponseEntity().getData()) {
+        updateEvent(organizationId, event);
+      }
+    } else {
+      logger.severe(String.format("Request list organization %s failed on [%d] %s", organizationId.toString(), response.getStatus(), response.getMessage()));
+    }
+  }
+
+  private void updateEvent(OrganizationId organizationId, Event mikkeliNytEvent) {
+    EventId mikkeliNytEventId = new EventId(organizationId, MikkeliNytConsts.IDENTIFIER_NAME, mikkeliNytEvent.getId());
+    
+    Identifier identifier = identifierController.findIdentifierById(mikkeliNytEventId);
+    if (identifier == null) {
+      identifier = identifierController.createIdentifier(mikkeliNytEventId);
+    }
+    
+    EventId kuntaApiId = new EventId(organizationId, KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+    fi.otavanopisto.kuntaapi.server.rest.model.Event event = translate(kuntaApiId, mikkeliNytEvent);
+    
+    if (StringUtils.isNotBlank(mikkeliNytEvent.getImage())) {
+      updateAttachment(organizationId, kuntaApiId, mikkeliNytEvent.getImage());
+    }
+    
+    modificationHashCache.put(kuntaApiId.getId(), createPojoHash(event));
+    eventCache.put(kuntaApiId, event);
+  }
+
+  private void updateAttachment(OrganizationId organizationId, EventId eventId, String imageUrl) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateAttachment" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)